api/urls.py

```diff
@@ -0,0 +1,18 @@
+from django.urls import path, include
+from . import views
+
+
+urlpatterns = [
+    path('api-auth/', include('rest_framework.urls')),
+    path('currently_playing/', views.currently_playing),
+    path('currently_playing_with_schedule/', views.currently_playing_with_schedule),
+    path('movie/<int:pk>/', views.MovieListCreate.as_view()),
+    path('movies/', views.MovieListCreate.as_view()),
+    path('hall/<int:pk>/', views.HallListCreateUpdate.as_view()),
```

Comment:
And yes.. I missed this piece of information before. This way, you still have to configure url for these routes individually. Plus, I think that it adds both methods for each endpoint; meaning that `halls/` can be called with both `GET` and `POST`. You can go one step further in the docs and read about viewsets https://www.django-rest-framework.org/api-guide/viewsets/
A nice feature is that they allow you to register the url config paths under a router, without adding the routes manually for each http method. They also use a standard config for the urls, which puts for another general rule instead of possible different url configs as each dev thinks different (which can help on a bigger project)

Reply:
Routers are nice!